app/helpers/events_helper.rb

```diff
@@ -12,8 +12,8 @@ def event_title_line(event)
   end
 
   def users_for_event(state)
-    req = @event.requests
-    req = req.where(state: state) if state != 'all'
-    user_email = req.map { |e| e.user.email }.uniq
+    requests = @event.travel_sponsorships.includes(:user).select(:user_id).distinct.accessible_by(current_ability)
+    requests = requests.where(state: state) if state != 'all'
+    requests.pluck(:'users.email')
```

Comment:
Why the colon ``:``? And I think :email should be enough as it is distinct, shouldn't it?